src/fabric.c

```diff
@@ -481,6 +481,31 @@ FI_DESTRUCTOR(fi_fini(void))
 	ofi_osd_fini();
 }
 
+void ofi_free_nic(struct fid_nic *nic)
+{
+	if (nic->device_attr) {
```

Comment:
Do you want to handle a `NULL` nic pointer in here somehow? 

Maybe wrap this in one more `if`-block, or maybe make all of these something like:
```c
if (nic && nic->device_attr) {
}
```